### PR TITLE
Redux refactor components

### PR DESCRIPTION
**Pull request by andreykishtov**
doing redux refactor.
easier to work on with pull request.
created homepage cards call async preperation.
created first stage filtered cards.

am i going the right way?

### CODE REVIEW

**Comment by andreykishtov** [comment]
filter function needs to be reworked its too complicated to redo for redux in this state

**Comment by andreykishtov** [comment]
filter component needs to be written from scratch its too complicated to redo for redux in this state

**Comment by Alaev** [comment]
@andreykishtov yep, I told you.
it would be best if you continue the work here on this branch and add a small todo list

File: src/components/pages/HomePage/HomePage.js [hunk 1]
```diff
@@ -19,10 +21,16 @@ class HomePage extends Component {
...
-    this.setState({ special });
+  componentWillMount = async () => {
+    try {
+      await this.props.dispatch(cardsRequest());
+      const { cards } = this.props;
+      const special = this.getRandomCard(cards);
+      await this.setState({ filteredCards: cards, special });
+      this.updatePagination();
+    } catch (error) {
+      console.log(error);
```
Comment (Alaev) [inline src/components/pages/HomePage/HomePage.js]: should use some pop notification with error like in home page

File: src/components/pages/HomePage/HomePage.js [hunk 2]
```diff
@@ -39,7 +47,7 @@ class HomePage extends Component {
     this.setState({ gridCards: selectedCard });
   };
 
-  getRandomCard = () => this.state.cards[Math.floor(Math.random() * this.state.cards.length)];
+  getRandomCard = arr => arr[Math.floor(Math.random() * arr.length)];
```
Comment (Alaev) [inline src/components/pages/HomePage/HomePage.js]: better to give better name then arr, an array of what? why? etc..

File: src/components/pages/HomePage/HomePage.js [hunk 3]
```diff
@@ -62,15 +70,16 @@ class HomePage extends Component {
...
+      toggleCTADialog: !this.state.toggleCTADialog
+    });
 
   render() {
-    const { special, filteredCards, cards, current, pageSize, gridCards } = this.state;
-
-    return (
+    const { special, filteredCards, current, pageSize, gridCards } = this.state;
+    const { cards } = this.props;
+    return filteredCards.length > 0 ? (
```
Comment (Alaev) [inline src/components/pages/HomePage/HomePage.js]: would be better to move this return into a function that will do the logic and return jsx

File: src/store/actions/homepage.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// import axios from 'axios';
```
Comment (Alaev) [inline src/store/actions/homepage.js]: comments

File: src/store/actions/cards.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// import axios from 'axios';
```
Comment (Alaev) [inline src/store/actions/cards.js]: better to remove comments

File: src/store/actions/cards.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+});
+
+// export const CardsRequest = (url = 'http://localhost') => async dispatch => {
+export const cardsRequest = () => async dispatch => {
+  dispatch(getCards());
+
+  try {
+    // const res = await axios.get(url);
+    const { cards } = data;
+    dispatch(getCardsSuccess(cards));
```
Comment (Alaev) [inline src/store/actions/cards.js]: I think we should try to load cards from local storage first,
let's say we store that cards for 2 days in local storage if they are older we will dispatch a new request.
would be also good option to look into Redux `Persist`

**Pull request merged**